src/launcher.ts

```diff
@@ -123,9 +189,27 @@ export default class VSCodeServiceLauncher extends ChromedriverServiceLauncher {
      * @returns "insiders" if `desiredReleaseChannel` is set to this otherwise a concrete version
      */
     private async _setupChromedriver (desiredReleaseChannel: string) {
-        const version = await this._fetchVSCodeVersion(desiredReleaseChannel)
+        let version = await this._fetchVSCodeVersion(desiredReleaseChannel)
 
         try {
+            /**
+             * if testing insiders fetch from latest release rather than from main
+             * branch given that there could be a situation where the VSCode team
+             * updates Electron but haven't released a stable VSCode version yet causing
+             * the service to download a wrong Chromedriver version
+             */
+            if (version === 'main') {
+                log.info(`Fetch latest GitHub release from ${VSCODE_LATEST_RELEASE}`)
+                const { body: versions } = await request(VSCODE_LATEST_RELEASE, {
+                    headers: {
+                        Accept: 'application/vnd.github.v3+json',
+                        'User-Agent': 'webdriverio-community/wdio-vscode-service'
+                    }
+                })
+                const response: { tag_name: string } = await versions.json()
+                version = response.tag_name
```

Comment:
@christian-bromann I just tested with v4. I have my config set with:
```
  capabilities: [
    {
      browserName: 'vscode',
      browserVersion: 'insiders',
...
```
but the stable version `1.69.2` is being returned. Does the above request include the insiders version or only stable releases?

Reply:
Or am I doing something wrong on my end?

Reply:
let me check